packages/wavs/src/subsystems/trigger/streams/atproto_jetstream.rs

```diff
@@ -0,0 +1,784 @@
+    pub record: Option<serde_json::Value>,
+    /// Repository revision identifier for this commit (if provided)
+    pub rev: Option<String>,
+    /// Index of the operation within the commit (0-based)
+    pub op_index: Option<u32>,
+}
+
+/// Create a Jetstream stream for ATProto events
+pub async fn start_jetstream_stream(
+    config: JetstreamConfig,
+    metrics: TriggerMetrics,
+) -> Result<Pin<Box<dyn Stream<Item = Result<StreamTriggers, TriggerError>> + Send>>, TriggerError>
+{
+    let stream = async_stream::stream! {
+        let mut reconnect_count = 0;
+        let max_reconnects = 10;
+        let base_delay = Duration::from_secs(1);
+        let max_delay = Duration::from_secs(60);
+
+        loop {
+            info!("Connecting to Jetstream at: {}", config.endpoint);
+
+            match create_jetstream_connection(&config, &metrics).await {
+                Ok(mut stream) => {
+                    reconnect_count = 0;
+                    info!("Successfully connected to Jetstream");
+
+                    while let Some(events) = stream.next().await {
+                        match events {
+                            Ok(atproto_events) => {
+                                for atproto_event in atproto_events {
+                                    yield Ok(StreamTriggers::AtProto {
+                                        event: atproto_event,
+                                    });
+                                }
+                            }
+                            Err(TriggerError::JetstreamParse(msg)) => {
+                                // Non-fatal parse issue (e.g. hello/keepalive)
+                                warn!("Ignoring Jetstream message: {}", msg);
+                                continue;
+                            }
+                            Err(e) => {
+                                error!("Error processing Jetstream event: {:?}", e);
+                                metrics.increment_total_errors("jetstream_event");
+                                break; // Break from inner loop to trigger reconnect
+                            }
+                        }
+                    }
+                }
+                Err(e) => {
+                    error!("Jetstream connection error: {:?}", e);
+                    metrics.increment_total_errors("jetstream_connection");
+
+                    if reconnect_count >= max_reconnects {
+                        error!("Max reconnection attempts reached, giving up");
+                        yield Err(TriggerError::JetstreamConnection("Max reconnection attempts reached".to_string()));
+                        return;
+                    }
+
+                    // Exponential backoff with jitter
+                    let delay = std::cmp::min(
+                        base_delay * 2_u32.pow(reconnect_count),
+                        max_delay
+                    ) + Duration::from_millis(rand::random::<u64>() % 1000);
```

Comment:
nit: 
```suggestion
                    ) + Duration::from_millis(rand::thread_rng().gen_range(0..1000));
```
IMO more readable plus `random::<u64>` can generate big numbers, on the other hand my suggestion requires an extra import. :shrug: